### ISSUE TITLE
Pyright does not respect submodules that are already imported in a module 

### ISSUE THREAD

**Issue by Zomatree** [microsoft/pyright#3340]
**Describe the bug**
python imports allow doing `folder.file` if `file` is already imported, however pyright is not detecting this when the file is imported like `from lib.file import ...`.

**To Reproduce**
*This is the simplest example i could think of*
Folder Tree:
```
.
├── lib
│   ├── b.py
│   └── __init__.py
└── test.py
``` 
`__init__.py`
```py
from lib.b import c

a = 1
```
`b.py`
```py
b = 2
c = 3
```
`test.py`
```py
import lib

print(lib.b.b)
```
![image](https://user-images.githubusercontent.com/39768508/163270302-98e79ba6-10d4-4ea1-94bd-780ccfc82cde.png)

**Expected behavior**
It should see that `lib.b` was already imported and should allow `lib.b.b`

**VS Code extension or command-line**
`pyright 1.1.237`

**Additional context**
This does work with `.b` instead of `lib.b`, however this isnt my code that im reporting this for and dont have the option to edit the code.

**Comment by erictraut** [comment]
This is intended behavior. Relying on cross-module loader side effects is not only difficult (impossible in many cases) to model statically, but it's also fragile at runtime. We therefore explicitly don't model it in pyright. 

If your intent is to access symbols from submodule `lib.b`, you should `import lib.b` from your module, not rely on the fact that `lib` implicitly loaded it as a side effect.
